<p>HelloWorld rule description.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class MyClass {
	
	int foo1(int value) { return 0; }
		
	MyClass foo2(MyClass value) { return null; }
	
	...
 
}
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/EYBUC">CERT, MSC11-J.</a> - Do not let session information leak within a servlet </li>
</ul>

